### PR TITLE
[feat]: Added Weather API

### PR DESCRIPTION
**Pull request by anjali1102**
image

### CODE REVIEW

**Review by shashikantmaurya0007 — APPROVED** [review]
code looks good to be merged

File: src/components/Weather/Weather.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import { WeatherBody } from "./WeatherBody";
+import "./WeatherBody.css";
+
+const Weather = () => {
+  const [lat, setLat] = useState([]);
+  const [long, setLong] = useState([]);
+  const [data, setData] = useState([]);
+
+  const REACT_APP_API_URL = "https://api.openweathermap.org/data/2.5";
+  const REACT_APP_API_KEY = "c962ed695546773fa54252b994d33923";
```
Comment (dev-enforced) [inline src/components/Weather/Weather.js]: Can use the env file for storing these credentials at a safe point by passing the .env in gitignore file and passing it to the deploy platform as well in order to make sure the app works on further previews!!

File: src/components/Weather/Weather.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        setLong(position.coords.longitude);
+      });
+
+      await fetch(
+        `${REACT_APP_API_URL}/weather/?lat=${lat}&lon=${long}&units=metric&APPID=${REACT_APP_API_KEY}`
+      )
+        .then((res) => res.json())
+        .then((result) => {
+          setData(result);
+          console.log(result);
```
Comment (dev-enforced) [inline src/components/Weather/Weather.js]: Can remove this debugger

File: src/components/Weather/Weather.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { useEffect, useState } from "react";
+import { WeatherBody } from "./WeatherBody";
+import "./WeatherBody.css";
+
+const Weather = () => {
+  const [lat, setLat] = useState([]);
+  const [long, setLong] = useState([]);
```
Comment (dev-enforced) [inline src/components/Weather/Weather.js]: Data type of lat and long as array seems to be not required here  can use string too.

**Review by dev-enforced — APPROVED** [review]
The live preview of Weather feature is working well. Liked the icon used to show the loading state till the weather info comes up!!

**Pull request merged**